Zend/zend_gc.c

```diff
@@ -147,6 +147,15 @@ static void gc_trace_ref(zend_refcounted *ref) {
 }
 #endif
 
+typedef struct _gc_refcounted_stack {
+	zend_refcounted *ref;
+	struct _gc_refcounted_stack *next;
+} gc_refcounted_stack;
```

Comment:
This is a very inefficient way to implement a stack, because it requires one allocation and deallocation for each element, and uses lots of indirection.

The usual way to implement a stack is to have a dynamically resized buffer and just maintain a pointer that moves up and down as elements are pushed / popped. This makes stack pushes/pops very cheap and amortized allocations.

Reply:
Yes, though about it but from wasn't sure how much slower it is. Do we have an implementation we can reuse in the codebase?

Also, is there a standard way to test performance of the different modules of the language, especially here GC module?

Reply:
done